velox/exec/tests/RowContainerTest.cpp

```diff
@@ -59,17 +68,59 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     }
   }
 
+  void testExtractColumnOffsetsForOddRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected) {
+    auto size = rows.size();
+
+    // Extract only odd rows.
+    // Test using extractColumnOffsets API.
+    auto oddsSize = size / 2;
+    auto offsetsBuffer =
+        AlignedBuffer::allocate<vector_size_t>(oddsSize, pool_.get());
+    offsetsBuffer->setSize(oddsSize * sizeof(vector_size_t));
+    auto rawOffsetsBuffer = offsetsBuffer->asMutable<vector_size_t>();
+    for (int i = 0; i < oddsSize; i++) {
+      rawOffsetsBuffer[i] = i * 2;
+    }
+
+    auto result = BaseVector::create(expected->type(), oddsSize, pool_.get());
+    container.extractColumnAtOffsets(
+        rows.data(), offsetsBuffer, column, 0, result);
+    EXPECT_EQ(oddsSize, result->size());
```

Comment:
It would be nice to avoid copy-pasting verification logic. Maybe  testExtractColumnXxxForOddRows methods could return the 'results' and the caller would verify it using a helper method?

Reply:
Previously the results were different. So I was using separate result verification.

Now I have changed the test to match results. So I have combined the verification logic also.